docs/pct/README.md

```diff
@@ -0,0 +1,235 @@
+
+Example usage:
+
+``` bash
+> cd /home/me/projects/MyBoltProject
+> pct new bolt-project
+
+> pct new puppet-module -n MyNewProject -o /home/me/projects/
+> cd /home/me/projects/MyNewProject
+> pct new puppet-fact -n ApplicationVersion
+> pct new rsapi-provider -n Awesomething
+> pct new puppet-transport -n AwesomethingApi
+```
+
+### Template Updates
+
+`pct new` will **NOT** update your code to a newer version of a template.
+
+If you run a `pct new` command using a `project` template, the project will replace the content within the output directory with the template code.
+
+If you run a `pct new` command using an `item` template, the item will suppliment the content within the output directory with the template code. If files / folders that are named the same as the template content already exist, it will overwite this content.
+
+## Writing Templates
+
+### Dos and Don'ts
+
+* `project` templates should provide all the code necessary to create a project from scratch and no more.
+* Do not include configuration files that can be added via an `item` template later by an end user, for example, CI job configuration.
+* Templates should be self documenting to help guide new users on how to use the file that has been created.
+
+### Structure
+
+A PCT must contain a `pct-config.yml` in the root directory, alongside a `content` directory. The root directory must be named the same as the `id` value defined in  your `pct-config.yml`
+
+The `content` directory contains the files and folders required to produce the `project` or `item`.
+
+To mark a file as a template, use the `.tmpl` extension. Templated files can also use the global variable of `{{pct_name}}` to access the input from the `--name` cli argument.
+
+> :memo: Folders within the `content` directory can also use the `{{pct_name}}` variable
+
+Example template file names:
+
+``` bash
+myConfig.json.tmpl
+{{pct_name}}_spec.rb
+```
+
+> :memo: One, all or none of the files can be templated.
+
+#### pct-config.yml
+
+Format of pct-config.yml
+
+``` yaml
+---
+template:
+  id: <a unique name>
+  type: <type>
+  display: <a human readable name>
+  version: <semver>
+  url: <url to project>
+
+<template parameters>
+```
+
+Example pct-config.yml:
+
+``` yaml
+---
+template:
+  id: example-template
+  type: project
+  display: Example
+  version: 0.1.0
+  url: https://github.com/puppetlabs/pct-example
+```
+
+Example structure for `example-template`:
+
+``` bash
+> tree ~/.pdk/pct/example-template-params
+/Users/me/.pdk/pct/example-template-params
+├── content
+│   └── example.txt.tmpl
+└── pct-config.yml
+```
+
+Example pct-config.yml with parameters:
+
+``` yaml
+---
+template:
+  id: example-template-params
+  type: project
+  display: Example with Parameters
+  version: 0.1.0
+  url: https://github.com/puppetlabs/pct-example-params
+
+
+example_template:
+  foo: "bar"
+  isPuppet: true
+  colours:
+  - "Red"
+  - "Blue"
+  - "Green"
+
+```
+
+### Templating Language
```

Comment:
Just a minor readability thing - given the above `yaml` snippet is related to the `go` snippet below, the **Templating Language** breaks up the flow a bit.

Reply:
I'm not sure there is any easy way to fix that other than copy/paste that code again.

Reply:
I had exactly the same suggestion as Ciaran. I hear your point about it kinda belonging in both places but the parameters topic is more related to the templating language section than it is for the file structure section. I vote for moving it entirely and not including it where it is at the moment